electrum/submarine_swaps.py

```diff
@@ -464,7 +464,7 @@ async def _claim_swap(self, swap: SwapData) -> None:
             if spent_height in [TX_HEIGHT_LOCAL, TX_HEIGHT_FUTURE]:
                 spent_height = None
             if spent_height is not None:
-                if spent_height > 0:
+                if spent_height > 0 and swap.preimage:
```

Comment:
What if we never learn the preimage?  Is it ok that this cleanup logic will never run?
For example, user starts a forward-swap, receives and holds LN-htlcs, funds the onchain script (only in mempool) and goes offline for long time. When he comes back again thousands of blocks later, he will timeout onchain the htlc. He will never learn the preimage.

Reply:
I see. Well okay.